File: EvaluateReversePolishNotation.java
class Solution {
    public int evalRPN(String[] tokens) {
        Stack<Integer> values = new Stack<>();
        for(int i = 0; i < tokens.length; i++){
            int val1 = 0;
            int val2 = 0;
            if(tokens[i].equals("/")){
                val1 = values.pop();
                val2 = values.pop();
                values.push(val2/val1);
            }
            else if(tokens[i].equals("-")){
                val1 = values.pop();
                val2 = values.pop();
                values.push(val2-val1);
            }
            else if(tokens[i].equals("+")){
                val1 = values.pop();
                val2 = values.pop();
                values.push(val2+val1);
            }
            else if(tokens[i].equals("*")){
                val1 = values.pop();
                val2 = values.pop();
                values.push(val2*val1);
            }
            else{
                int val = Integer.valueOf(tokens[i]); 
                values.push(val);
            }
        }
        return values.pop();
    }
}

File: EvaluateReversePolishNotation.py
class Solution(object):
    def evalRPN(self, tokens):
        """
        :type tokens: List[str]
        :rtype: int
        """
        nums = []
        for i in tokens:
            if (i == "+" or i == "-" 
            or i == "/" or i == "*"):
                val2 = nums.pop()
                val1 = nums.pop()
                comb = 0
                if(i == "+"):
                    comb = val1 + val2
                elif(i == "-"):
                    comb = val1 - val2
                elif(i == "/"):
                    comb = int(float(val1) / val2)
                else:
                    comb = val1 * val2
                nums.append(comb)
            else:
                nums.append(int(i))
        return nums.pop()
